### PR TITLE
MINOR: remove idempotent statement

### PR DESCRIPTION
**Pull request by mjsax**
Compare https://github.com/apache/kafka/pull/5657/files#r218147757

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @guozhangwang @bbejeck @vvcephei

**Comment by mjsax** [comment]
Retest this please

**Review by vvcephei — APPROVED** [review]
Looks right to me.

**Review by bbejeck — APPROVED** [review]
LGTM. Thanks @mjsax.

**Comment by bbejeck** [comment]
failures unrelated

retest this please

**Comment by mjsax** [comment]
Retest this please

**Comment by mjsax** [comment]
@guozhangwang Can you have a look? Java 10 does not build for `0.11.0` branch IMHO. Java8 passed.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java [hunk 1]
```diff
@@ -180,7 +180,6 @@ private void startRestoration(final Map<TopicPartition, StateRestorer> initializ
                             "Reinitializing the task and restore its state from the beginning.", task.id, restorer.storeName(), restorer.partition());
 
                         needsInitializing.remove(restoringPartition);
-                        initialized.put(restoringPartition, restorer);
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java:183]: Should we change it to `initialized.remove(partition);`  instead?

Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java:183]: NVM, saw your other comment on the paring PR.

**Review by guozhangwang — APPROVED** [review]

**Pull request merged**